genesis-tools/contracts/vesting-contract/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+//! A smart contract that allows tokens vesting and lockup.
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use key_management::{KeyType, PublicKey};
+use near_bindgen::{env, near_bindgen, Promise};
+use uint::construct_uint;
+
+construct_uint! {
+    /// 256-bit unsigned integer.
+    pub struct U256(4);
+}
+
+mod key_management;
+// mod lockup_vesting_transfer_rules;
+
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct VestingContract {
+    lockup_amount: u128,
+    lockup_timestamp: u64,
```

Comment:
What is the difference between `lockup_timestamp` and `vesting_end_timestamp `?
Need doc comments..